crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -36,6 +36,7 @@ pub(super) struct SemanticIndexBuilder<'db> {
     scope_stack: Vec<FileScopeId>,
     /// The assignment we're currently visiting.
     current_assignment: Option<CurrentAssignment<'db>>,
+    pattern_state: Option<MatchPatternState<'db>>,
```

Comment:
Any recommendation for the name here (`MatchPatternState`)?

Reply:
I'm leaning towards naming it `CurrentPattern` to match `current_assignment`. It makes it clear that both follow the same schema.